rfcs/20190506-filesystem-plugin-modular-tensorflow.md

```diff
@@ -0,0 +1,1082 @@
+# Modular Filesystems C API
+
+| Status        | Proposed                                             |
+:-------------- |:---------------------------------------------------- |
+| **Author(s)** | Mihai Maruseac (mihaimaruseac@google.com)            |
+| **Sponsor**   | Gunhan Gulsoy (gunan@google.com)                     |
+| **Updated**   | 2019-05-06                                           |
+
+## Abstract
+
+TensorFlow has grown large since its introduction and we are currently working
+on ensuring a better development workflow by a team-wide effort designed to make
+TensorFlow more modular according to the [current RFC][modular_rfc].
+
+A major module of the modular TensorFlow involves all operations regarding files
+and the filesystem. Saving models, checkpointing, reading files as input to
+models (i.e. images, audio, etc.), and many other file manipulations are all
+tasks that TensorFlow has to support.
+
+While users can use the underlying operating system’s API (such as calling fopen
+and friends on Posix environments, for example), it is definitely better to have
```

Comment:
nit: I'd mention cloud environments here as many people's thoughts when reading about filesystems will be around things like ext4 and xfs for which we do not need a layer

Reply:
Done.